myTrip/static/src/checkpoints.js

```diff
@@ -0,0 +1,166 @@
+		const latitude = 2;
+		const title = 'test6';
+		const description = 'test';
+		const position_number = 1;
+		const source_url = '';
+		axios.post('/api/v1/trip/11/checkpoint/', {
+			longitude,
+			latitude,
+			title,
+			description,
+			position_number,
+			source_url})
+		.then(function(response){
+			console.log(response.data);
+		})
+	}
+
+	render(){
+		return(
+			<div>
+				<RaisedButton label="Add Checkpoint" onClick={this.CreateCheckpoint}/>
+			</div>
+		);
+	}
+}
+
+class DeleteCheckpoint extends React.Component{
+
+	Delete (event) {
+		axios.delete('/api/v1/trip/11/checkpoint/9/')
+		.then(function(response){
+			console.log(response.data)
+		})
+	}
+
+	render(){
+		return(
+			<div>
+				<RaisedButton label="Delete Checkpoint"  onClick={this.Delete}/>
+			</div>
+		);
+	}
+}
+
+class GetCheckpoints extends React.Component{
+
+	constructor(props) {
+		super(props);
+		this.state = {list:[]};
+		this.GetAll = this.GetAll.bind(this);
+	}
+
+	GetAll (event) {
+		var self = this
+		axios.get('/api/v1/trip/11/checkpoint/')
+		.then(function(response){
+			// console.log(response.data);
+			self.setState({list:response.data})
+		})
+	}
+	// GetAll (event) {
+	// 	axios.get('/api/v1/trip/11/checkpoint/')
+	// 	.then((response) => {
+	// 		console.log(response.data);
+	// 		this.setState({list: response.data})
+	// 	})
+	// }
+
+	render(){
+		return(
+			<div>
+				<RaisedButton label="Get Checkpoints" onClick={this.GetAll}/>
+				<CheckpointList list = {this.state.list}/>
+			</div>
+		);
+	}
+}
+
+class CheckpointList extends React.Component{
+
+	render(){
+		// console.log(this.props.list)
+		var list = this.props.list;
+		if(list.length > 0){
+			// list = list.reverse();
+			list = list.map(item =>{
+				console.log(item);
+				 return (
+					<CheckpoinItem key={item.id} list={item} />
+				);
+			})
+			console.log(list)
+			return(
+				<div>
+					{list}
+				</div>);
+		}
+		else{
+			return(
+				<div>
+				</div>
+			);
+		}
+	}
+}
+
+class CheckpoinItem extends React.Component {
+	constructor(props) {
+		super(props);
+		this.details = this.details.bind(this);
+		this.delete = this.delete.bind(this);
+	}
+
+	details(event){
+		alert(this.props.list.id);
+	}
+
+	delete (event) {
+		var checkpoint_id = this.props.list.id;
+		axios.delete('/api/v1/trip/11/checkpoint/'+checkpoint_id+'/')
+		.then(function(response){
+			console.log(response)
+		})
+	}
+
+	render() {
+		return (
+			<div>
+				<button onClick={this.details}>id: {this.props.list.id} - 
+				title: {this.props.list.title}</button>
+				<button onClick={this.delete}>delete</button>
+			</div>
+		);
+	}
+}
+
+class CheckpoinDetails extends React.Component {
+	constructor(props) {
+		super(props);
+	}
+
+	render() {
+		console.log('details props',this.props)
+		return (
+			<div>
+			</div>
+		);
+	}
+}
+
+module.exports = {
+	AddCheckpoint: AddCheckpoint,
+	DeleteCheckpoint: DeleteCheckpoint,
+	GetCheckpoints: GetCheckpoints,
+	CheckpoinDetails: CheckpoinDetails
+}
```

Comment:
add new line and push funcition which makes AJAX requests to checkpoint.service.js